packages/flutter/lib/src/widgets/focus_scope.dart

```diff
@@ -429,14 +434,33 @@ class _FocusState extends State<Focus> {
         widget.onFocusChange(focusNode.hasFocus);
       }
     }
+    if (_hasPrimaryFocus != focusNode.hasPrimaryFocus) {
+      setState(() {
+        _hasPrimaryFocus = focusNode.hasPrimaryFocus;
+      });
+    }
+    if (_canRequestFocus != focusNode.canRequestFocus) {
+      setState(() {
+        _canRequestFocus = focusNode.canRequestFocus;
+      });
+    }
+    if (_skipTraversal != focusNode.skipTraversal) {
```

Comment:
What's skipTraversal used for? Seems like its only set, but never really used?

Reply:
It's a property of focus nodes, where it will be skipped for focus traversal purposes, but can still take the primary focus if given it explicitly.  It differs from `canRequstFocus` in that `canRequestFocus` prevents it from ever getting primary focus AND being traversed.